File: pset6/readability/readability.py
from cs50 import get_string


def main():
    text = get_string("Text: ")

    letters = count_letters(text)
    words = count_words(text)
    sentences = count_sentences(text)
    grade = calculate_grade(letters, words, sentences)
    print_grade(grade)


# Counts text's letters
def count_letters(text):
    counter = 0
    for i in range(len(text)):

        # Checks for letters
        if (text[i] >= 'A' and text[i] <= 'Z') or (text[i] >= 'a' and text[i] <= 'z'):
            counter += 1

    return counter


# Counts text's words
def count_words(text):
    counter = 0
    for i in range(len(text)):

        # Checks for empty spaces signaling the word's end
        if text[i] == ' ':
            counter += 1

    counter += 1
    return counter


# Counts text's sentences
def count_sentences(text):
    counter = 0
    for i in range(len(text)):

        # Checks for punctiation signaling the sentence's end
        if text[i] == '.' or text[i] == '!' or text[i] == '?':

            counter += 1

    return counter


# Calculates text's grade
def calculate_grade(letters, words, sentences):

    # Calculates the average of letters per 100 words
    l = letters / words * 100
    # Calculates the average of sentences per 100 words
    s = sentences / words * 100
    # The Coleman-Liau index rounded
    index = round(0.0588 * l - 0.296 * s - 15.8)

    return index


# Prints grade with the right formatting
def print_grade(grade):
    if grade < 1:
        print("Before Grade 1")
    elif grade > 16:
        print("Grade 16+")
    else:
        print(f"Grade {grade}")


main()

File: pset2/readability/readability.c
>#include <ctype.h>
#include <cs50.h>
#include <stdio.h>
#include <string.h>
#include <math.h>

int count_letters(string text);
int count_words(string text);
int count_sentences(string text);
int calculate_grade(int letters, int words, int sentences);
void print_grade(int grade);


//Main function. Prompts the user for a text, and returns its reading grade level in the north american standard
int main(void)
{
    string text = get_string("Text: ");

    int letters = count_letters(text);
    int words = count_words(text);
    int sentences = count_sentences(text);
    int grade = calculate_grade(letters, words, sentences);
    print_grade(grade);
}


//Diferentiates letters from other symbols and counts them
int count_letters(string text)
{
    int counter = 0;
    for (int i = 0, n = strlen(text); i < n; i++)
    {
        if ((text[i] >= 'A' && text[i] <= 'Z') || (text[i] >= 'a' && text[i] <= 'z'))
        {
            counter++;
        }
        else
        {

        }
    }
    return counter;
}


//Counts words by calculating the number of spaces
int count_words(string text)
{
    int counter = 0;
    for (int i = 0, n = strlen(text); i < n; i++)
    {
        if (text[i] == ' ')
        {
            counter++;
        }
    }
    counter++;
    return counter;
}


//Counts sentences by calculating the number of punctuation marks
int count_sentences(string text)
{
    int counter = 0;
    for (int i = 0, n = strlen(text); i < n; i++)
    {
        if (text[i] == '.' || text[i] == '!' || text[i] == '?')
        {
            counter++;
        }
    }
    return counter;
}


//Calculates the grade trough the Coleman-Liau index and rounds it.
int calculate_grade(int letters, int words, int sentences)
{
    //Calculates the average of letters per 100 words
    float l = (float) letters / words * 100;
    //Calculates the average of sentences per 100 words
    float s = (float) sentences / words * 100;
    //The Coleman-Liau index rounded
    float index = round(0.0588 * l - 0.296 * s - 15.8);

    return index;
}


//Correctly prints the grades, from Before grade 1 to Grade 16+
void print_grade(int grade)
{
    if (grade < 1)
    {
        printf("Before Grade 1\n");
    }
    else if (grade > 16)
    {
        printf("Grade 16+\n");
    }
    else
    {
        printf("Grade %i\n", grade);
    }
}